packages/sdk/src/modules/loggerModule/index.ts

```diff
@@ -1,14 +1,28 @@
 import { LoggerFactory, LoggerType } from "@vue-storefront/logger";
 import type { Module } from "../../types";
 import type { LoggerModuleConfig } from "./types";
+import { wrapLogger } from "./wrapper";
+import { isInvalidConfig } from "./utils";
 
 export const loggerModule = (options?: LoggerModuleConfig) => {
   const logger =
     options?.handler ?? LoggerFactory.create(LoggerType.ConsolaGcp, options);
 
+  if (isInvalidConfig(options)) {
+    console.warn(
+      "Both handler and options are provided. Using handler, options will be ignored."
+    );
+  }
+
+  const wrappedLogger = wrapLogger(logger, {
+    alokai: {
+      context: "storefront",
+    },
+  });
+
   return {
     connector: {
-      ...logger,
+      ...wrappedLogger,
```

Comment:
We can't use spread operator here as wrapLogger/injectMetadata wraps logger in Proxy's instance. And spread destroys Proxy.

Reply:
Actually it works pretty well :)

Reply:
I was testing it before and after your comment @Fifciu . I've created a snippet to reproduce this behavior -> https://www.typescriptlang.org/play/?#code/MYewdgzgLgBFCGAnA5gU1gXhgbxvAXDAIwA0MARoQExnCEDMMAvgNwBQoksAFvGACYAbVIhhZsbGDDRRCAChDkAVvj4BPMgAdEITarBqAlGIB8OSVJicIIYQDpBIZHIAGAQWDBUECAEswyDDauiJQajAAJNjBmkwuhuyWMIjoAK6IYDCKSgDaMQC6MADUMADkAPrBAB5q5aXFMAAi8FCodmAgAO5yCRZMJGysbBzg0EE6NWIwYKidMAAKE2pyCCjoZLwCwoi9I5C2bY7O1Wp28LsA9BdWo7AQ2qjw-ADyyqjAmDgwdj8nzOxsK43fb2I5ye4pJ6vJTvKB2ci9CDoAAqvgAtqgQKkoHIeqZzElrLBcMBmFMTolLNYDg4nHJgL1+lQAAyswxsIA

Reply:
On the surface, it indeed looks like it works but if you dive deeper you can spot it doesn't, e.g.: https://www.typescriptlang.org/play/?#code/MYewdgzgLgBFCGAnA5gU1gXhgbxvAXDAIwA0MARoQExnCEDMMAvgNwBQoksAFvGACYAbVIhhZsbGDDRRCAChDkAVvj4BPMgAdEITarBqAlGIB8OSVJicIIYQDpBIZHIAGAERABrcAFclqGH4QaFRNHxhtAC9USIidAA81GAASbG1dJhdDdksYRHQfRDAYRSUAbXTNAF0YAGoYAHIAfXTEpoa6mDd4KFQ7MBAAdzlsiyYSNlY2DnBoOJBEsRgwVEGYAAUEtTkEFHQyXgFhRFHJa1gIbVR4fgB5ZVRgTBwYOzfWpNYYAHpvmAAJEQBUqPHjwCAUVCoYrARwrfh4AQwACWUAaEIGsA+iLUAFsQPk7DAAMJwsDIsDIebAVAQCFQRDI5BofIIgBEMjZJWKqAAbiIkp5UEkoLw0RDBtwkoMAhAoTB6PE8rSfIJYCAAGZWWa2PqOZBnHX2fVyS75G73fxPOzwbI-P63GFk1AIkFPMgyvDAGl0mC8+CCHzArVs+BcyoiKBJPgI6y6hxOFEYkCwBlMlkunH4-JkcEwM3XO4PJ5J5YpvDzRbqbN9SS7GQ2pYAVgADC2WPaYAAZdDomD4-jIjVJf2B4MlYuwHwCESVtSGyDxk0Fi2Tm1234wACCEJXRatYIh5ChxXyNkE-IRFP7IFx0PVWth4HJlMRV-FZdg8DnByBME9MjEC0WxNN0vT9EMIxkJiMCti2TQAFQAIQLueepOHIHzrh2m5bki5qCHO+ZQMigiEcgyL8hCeZwcBCxqKBPR9AMwyGJIQA